application/common/models/User.php

```diff
@@ -343,6 +343,55 @@ private function savePassword()
         return true;
     }
 
+    public static function getExpiringUsers($criteria): array
+    {
+        if (empty($criteria)) {
+            return [];
+        }
+
+        $users = User::find()->joinWith('currentPassword')
+                             ->where(['active' => 'yes']);
+
+        foreach ($criteria as $name => $value) {
+            switch ($name) {
+                case 'expires_on':
+                case 'grace_period_ends_on':
+                    $users->andWhere(["password.$name" => $value]);
+                    break;
+                default:
+                    // if no criteria names match, this will ensure an empty result is returned
+                    $users->where('0=1');
+            }
+        }
+
+        return $users->all();
+    }
+
+    public static function getNewUsers($createdOn): array
```

Comment:
@forevermatt that is the desired behavior. The "new" concept is intended for triggering a welcome email and we want that to happen as close to when the user first became aware of their account and used it. We don't want to trigger emails when they are first created in the DB since that could be at least weeks before it is announced to users and an automated email would look like a phishing attack. We want to keep the context of creating a password and receiving a welcome email close together.